src/modules/process-input/next-string-result.interface.ts

```diff
@@ -0,0 +1,5 @@
+export interface NextStringResult {
+    done: boolean;
+    index: number;
+    str: string;
```

Comment:
what is str? let's always try to avoid abbreviations and be more explicit with our properties/variables

Reply:
updated to nextString: see commit